tests/Unit/Adapter/Order/GeneralConfigurationTest.php

```diff
@@ -0,0 +1,242 @@
+
+        $this->mockShopConfiguration
+            ->method('getShopConstraint')
+            ->willReturn($shopConstraint);
+
+        $this->mockConfiguration
+            ->method('get')
+            ->willReturnMap(
+                [
+                    ['PS_FINAL_SUMMARY_ENABLED', false, $shopConstraint, true],
+                    ['PS_GUEST_CHECKOUT_ENABLED', false, $shopConstraint, true],
+                    ['PS_DISALLOW_HISTORY_REORDERING', false, $shopConstraint, true],
+                    ['PS_PURCHASE_MINIMUM', null, $shopConstraint, self::MINIMUM_PURCHASE_VALUE],
+                    ['PS_ORDER_RECALCULATE_SHIPPING', false, $shopConstraint, true],
+                    ['PS_ALLOW_MULTISHIPPING', false, $shopConstraint, true],
+                    ['PS_SHIP_WHEN_AVAILABLE', false, $shopConstraint, true],
+                    ['PS_CONDITIONS', false, $shopConstraint, true],
+                    ['PS_CONDITIONS_CMS_ID', null, $shopConstraint, self::TOS_CMS_ID],
+                ]
+            );
+
+        $result = $generalConfiguration->getConfiguration();
+        $this->assertSame(
+            [
+                'enable_final_summary' => true,
+                'enable_guest_checkout' => true,
+                'disable_reordering_option' => true,
+                'purchase_minimum_value' => self::MINIMUM_PURCHASE_VALUE,
+                'recalculate_shipping_cost' => true,
+                'allow_multishipping' => true,
+                'allow_delayed_shipping' => true,
+                'enable_tos' => true,
+                'tos_cms_id' => self::TOS_CMS_ID,
+            ],
+            $result
+        );
+    }
+
+    /**
+     * @dataProvider provideInvalidConfiguration
+     *
+     * @param string $exception
+     * @param array $values
+     */
+    public function testUpdateConfigurationWithInvalidConfiguration(string $exception, array $values): void
+    {
+        $generalConfiguration = new GeneralConfiguration($this->mockConfiguration, $this->mockShopConfiguration, $this->mockMultistoreFeature);
+
+        $this->expectException($exception);
+        $generalConfiguration->updateConfiguration($values);
+    }
+
+    /**
+     * @return array[]
+     */
+    public function provideInvalidConfiguration(): array
+    {
+        return [
+            [UndefinedOptionsException::class, ['does_not_exist' => 'does_not_exist']],
+            [InvalidOptionsException::class, [
+                'enable_final_summary' => 'wrong_type',
+                'enable_guest_checkout' => true,
+                'disable_reordering_option' => true,
+                'purchase_minimum_value' => self::MINIMUM_PURCHASE_VALUE,
+                'recalculate_shipping_cost' => true,
+                'allow_multishipping' => true,
+                'allow_delayed_shipping' => true,
+                'enable_tos' => true,
+                'tos_cms_id' => self::TOS_CMS_ID,
+            ]],
+            [InvalidOptionsException::class, [
```

Comment:
not sure if there's a way to test wrong types for all fields at once ? 🤔

Reply:
Probably better to test invalid field one by one Or you can't be sure which field raised the exception
Though I believe the configuration services can handle partial data now no? So maybe you don't need to provide a full array, just the invalid field should be enough and easier to read